raiden-ts/src/raiden/helpers.ts

```diff
@@ -0,0 +1,179 @@
+import goerliDeploy from '../deployment/deployment_goerli.json';
+import ropstenServicesDeploy from '../deployment/deployment_services_ropsten.json';
+import rinkebyServicesDeploy from '../deployment/deployment_services_rinkeby.json';
+import goerliServicesDeploy from '../deployment/deployment_services_goerli.json';
+
+/**
+ * Returns contract information depending on the passed [[Network]]. Currently, only
+ * `rinkeby`, `ropsten` and `goerli` are supported.
+ * Throws an exception if called with another [[Network]].
+ *
+ * @param network - an account used for signing
+ * @returns deployed contract information of the network
+ */
+export const getContracts = (network: Network): ContractsInfo => {
+  switch (network.name) {
+    case 'rinkeby':
+      return ({
+        ...rinkebyDeploy.contracts,
+        ...rinkebyServicesDeploy.contracts,
+      } as unknown) as ContractsInfo;
+    case 'ropsten':
+      return ({
+        ...ropstenDeploy.contracts,
+        ...ropstenServicesDeploy.contracts,
+      } as unknown) as ContractsInfo;
+    case 'goerli':
+      return ({
+        ...goerliDeploy.contracts,
+        ...goerliServicesDeploy.contracts,
+      } as unknown) as ContractsInfo;
+    default:
+      throw new Error(
+        `No deploy info provided nor recognized network: ${JSON.stringify(network)}`,
+      );
+  }
+};
+
+/**
+ * Returns a [[Signer]] based on the `account` and `provider`.
+ * Throws an exception if the `account` is not a valid address or private key.
+ *
+ * @param account - an account used for signing
+ * @param provider - a provider
+ * @returns a [[Signer]] or [[Wallet]] that can be used for signing
+ */
+export const getSigner = async (account: string | number | Signer, provider: JsonRpcProvider) => {
+  let signer;
+
+  if (Signer.isSigner(account)) {
+    if (account.provider === provider) {
+      signer = account;
+    } else if (account instanceof Wallet) {
+      signer = account.connect(provider);
+    } else {
+      throw new Error(`Signer ${account} not connected to ${provider}`);
+    }
+  } else if (typeof account === 'number') {
+    // index of account in provider
+    signer = provider.getSigner(account);
+  } else if (Address.is(account)) {
+    // address
+    const accounts = await provider.listAccounts();
+    if (!accounts.includes(account)) {
+      throw new Error(`Account "${account}" not found in provider, got=${accounts}`);
+    }
+    signer = provider.getSigner(account);
+  } else if (PrivateKey.is(account)) {
+    // private key
+    signer = new Wallet(account, provider);
+  } else {
+    throw new Error('String account must be either a 0x-encoded address or private key');
+  }
+
+  return signer;
+};
+
+/**
+ * Initializes the [[transfers$]] observable
+ *
+ * @param state$ - Observable of the current RaidenState
+ * @returns observable of sent and completed Raiden transfers
+ */
+export const initTransfersObservable = (
```

Comment:
@andrevmatos should this also follow the `name$` pattern?

Reply:
It'd be nice yes to kepe the pattern, `initTransfers$` as a function which returns an observable